### PR TITLE
Executable lines and coverage rework (DO NOT MERGE)

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/ast/visitors/ExecutableLinesVisitor.java [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+      }
+    }
+  }
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    super.scanFile(context);
+
+    FileLinesContext fileLinesContext = sonarComponents.fileLinesContextFor(context.getFile());
+    // TODO minimize access to files, another one in LinesVisitor
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/ast/visitors/ExecutableLinesVisitor.java]: INFO Complete the task associated to this TODO comment. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1135)

File: java-jacoco/src/main/java/org/sonar/plugins/jacoco/JacocoConstants.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  private JacocoConstants() {
+  }
+
+  public static List<PropertyDefinition> getPropertyDefinitions(Version sqVersion) {
+    String subCategory = "JaCoCo";
+    if (sqVersion.isGreaterThanOrEqual(SQ_6_2)) {
+      return ImmutableList.of(
+        PropertyDefinition.builder(JacocoConstants.REPORT_PATHS_PROPERTY)
+          .defaultValue(JacocoConstants.REPORT_PATHS_DEFAULT_VALUE)
+          .category(CoreProperties.CATEGORY_JAVA)
```
Comment (SonarTech) [inline java-jacoco/src/main/java/org/sonar/plugins/jacoco/JacocoConstants.java]: MINOR Remove this use of "CATEGORY_JAVA"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: java-jacoco/src/main/java/org/sonar/plugins/jacoco/JacocoConstants.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          .defaultValue(JacocoConstants.REPORT_PATH_DEFAULT_VALUE)
+          .category(CoreProperties.CATEGORY_JAVA)
+          .subCategory(subCategory)
+          .name("UT JaCoCo Report")
+          .description("Path to the JaCoCo report file containing coverage data by unit tests. The path may be absolute or relative to the project base directory.")
+          .onQualifiers(Qualifiers.PROJECT, Qualifiers.MODULE)
+          .build(),
+        PropertyDefinition.builder(JacocoConstants.IT_REPORT_PATH_PROPERTY)
+          .defaultValue(JacocoConstants.IT_REPORT_PATH_DEFAULT_VALUE)
+          .category(CoreProperties.CATEGORY_JAVA)
```
Comment (SonarTech) [inline java-jacoco/src/main/java/org/sonar/plugins/jacoco/JacocoConstants.java]: MINOR Remove this use of "CATEGORY_JAVA"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: java-jacoco/src/main/java/org/sonar/plugins/jacoco/JacocoConstants.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+          .subCategory(subCategory)
+          .name("IT JaCoCo Report [DEPRECATED]")
+          .description("Deprecated since SonarQube 6.2. Use '" + JacocoConstants.REPORT_PATHS_PROPERTY + "' instead.")
+          .onQualifiers(Qualifiers.PROJECT, Qualifiers.MODULE)
+          .build());
+    } else {
+      return ImmutableList.of(
+        PropertyDefinition.builder(JacocoConstants.REPORT_PATH_PROPERTY)
+          .defaultValue(JacocoConstants.REPORT_PATH_DEFAULT_VALUE)
+          .category(CoreProperties.CATEGORY_JAVA)
```
Comment (SonarTech) [inline java-jacoco/src/main/java/org/sonar/plugins/jacoco/JacocoConstants.java]: MINOR Remove this use of "CATEGORY_JAVA"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: java-jacoco/src/main/java/org/sonar/plugins/jacoco/JacocoConstants.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+          .subCategory(subCategory)
+          .name("IT JaCoCo Report")
+          .description("Path to the JaCoCo report file containing coverage data by integration tests. The path may be absolute or relative to the project base directory.")
+          .onQualifiers(Qualifiers.PROJECT, Qualifiers.MODULE)
+          .build(),
+        PropertyDefinition.builder(JacocoConstants.REPORT_MISSING_FORCE_ZERO)
+          .defaultValue(Boolean.toString(JacocoConstants.REPORT_MISSING_FORCE_ZERO_DEFAULT_VALUE))
+          .name("Force zero coverage")
+          .category(CoreProperties.CATEGORY_JAVA)
```
Comment (SonarTech) [inline java-jacoco/src/main/java/org/sonar/plugins/jacoco/JacocoConstants.java]: MINOR Remove this use of "CATEGORY_JAVA"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: java-jacoco/src/main/java/org/sonar/plugins/jacoco/AbstractAnalyzer.java [hunk 6]
```diff
@@ -273,6 +264,7 @@ private static void analyzeFile(NewCoverage newCoverage, InputFile resource, ISo
 
   protected abstract CoverageType coverageType();
```
Comment (SonarTech) [inline java-jacoco/src/main/java/org/sonar/plugins/jacoco/AbstractAnalyzer.java]: MINOR Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 15 issues
* MINOR 14 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [AbstractAnalyzer.java#L138](https://github.com/SonarSource/sonar-java/blob/f4a0035e556b70d62e862fcf673f74c9f76dd53d/java-jacoco/src/main/java/org/sonar/plugins/jacoco/AbstractAnalyzer.java#L138): Remove this use of "ofType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoItSensor.java#L78](https://github.com/SonarSource/sonar-java/blob/f4a0035e556b70d62e862fcf673f74c9f76dd53d/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoItSensor.java#L78): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoItSensor.java#L79](https://github.com/SonarSource/sonar-java/blob/f4a0035e556b70d62e862fcf673f74c9f76dd53d/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoItSensor.java#L79): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoOverallSensor.java#L79](https://github.com/SonarSource/sonar-java/blob/f4a0035e556b70d62e862fcf673f74c9f76dd53d/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoOverallSensor.java#L79): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoOverallSensor.java#L80](https://github.com/SonarSource/sonar-java/blob/f4a0035e556b70d62e862fcf673f74c9f76dd53d/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoOverallSensor.java#L80): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoSensor.java#L145](https://github.com/SonarSource/sonar-java/blob/f4a0035e556b70d62e862fcf673f74c9f76dd53d/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoSensor.java#L145): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoSensor.java#L146](https://github.com/SonarSource/sonar-java/blob/f4a0035e556b70d62e862fcf673f74c9f76dd53d/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoSensor.java#L146): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 18 issues
* MINOR 17 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [AbstractAnalyzer.java#L144](https://github.com/SonarSource/sonar-java/blob/c101837236dda65877f1c7a7633aaf30be645577/java-jacoco/src/main/java/org/sonar/plugins/jacoco/AbstractAnalyzer.java#L144): Remove this use of "ofType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoItSensor.java#L78](https://github.com/SonarSource/sonar-java/blob/c101837236dda65877f1c7a7633aaf30be645577/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoItSensor.java#L78): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoItSensor.java#L79](https://github.com/SonarSource/sonar-java/blob/c101837236dda65877f1c7a7633aaf30be645577/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoItSensor.java#L79): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoItSensor.java#L79](https://github.com/SonarSource/sonar-java/blob/c101837236dda65877f1c7a7633aaf30be645577/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoItSensor.java#L79): Remove this use of "IT"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoOverallSensor.java#L79](https://github.com/SonarSource/sonar-java/blob/c101837236dda65877f1c7a7633aaf30be645577/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoOverallSensor.java#L79): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoOverallSensor.java#L80](https://github.com/SonarSource/sonar-java/blob/c101837236dda65877f1c7a7633aaf30be645577/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoOverallSensor.java#L80): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoOverallSensor.java#L80](https://github.com/SonarSource/sonar-java/blob/c101837236dda65877f1c7a7633aaf30be645577/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoOverallSensor.java#L80): Remove this use of "OVERALL"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoSensor.java#L145](https://github.com/SonarSource/sonar-java/blob/c101837236dda65877f1c7a7633aaf30be645577/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoSensor.java#L145): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoSensor.java#L146](https://github.com/SonarSource/sonar-java/blob/c101837236dda65877f1c7a7633aaf30be645577/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoSensor.java#L146): Remove this use of "CoverageType"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. MINOR [JaCoCoSensor.java#L146](https://github.com/SonarSource/sonar-java/blob/c101837236dda65877f1c7a7633aaf30be645577/java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoSensor.java#L146): Remove this use of "UNIT"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by benzonico** [comment]
Dealt with in #1132 Thanks.

**Pull request closed without merging**